image-rauc.c

```diff
@@ -91,11 +96,14 @@ static int rauc_generate(struct image *image)
 			return ret;
 	}
 
-	systemp(image, "rm -f '%s'", imageoutfile(image));
+	if (keyring)
+		xasprintf(&keyringarg, "--keyring='%s'", keyring);
 
-	ret = systemp(image, "%s bundle '%s' --cert='%s' --key='%s' %s '%s'",
+	systemp(image, "rm -f '%s'", imageoutfile(image));
+	ret = systemp(image, "%s bundle '%s' --cert='%s' --key='%s' %s %s '%s'",
 			get_opt("rauc"), tmpdir, cert, key,
-			extraargs, imageoutfile(image));
+			keyring ? keyringarg : "", extraargs,
```

Comment:
Maybe initialize `keyringarg` with `""` so you don't need the condition here?

Reply:
Done.